src/libraries/Microsoft.PowerFx.Connectors/Shared/ServiceFunction.cs

```diff
@@ -46,18 +47,27 @@ internal sealed class ServiceFunction : BuiltinFunction, IAsyncTexlFunction2
         private readonly WeakReference<IService> _parentService;
         private readonly string _actionName;
         private readonly bool _numberIsFloat;
+        private readonly string _pageLink;
+        private readonly bool _isDeprecated;
+        private readonly bool _isSupported;
+        private readonly string _notSupportedReason;
+        private readonly int _maxRows;
         internal readonly ServiceFunctionParameterTemplate[] _requiredParameters;
 
         public IEnumerable<TypedName> OptionalParams => _optionalParamInfo.Values;
         public Dictionary<string, TypedName> OptionalParamInfo => _optionalParamInfo;
         public override Capabilities Capabilities => Capabilities.OutboundInternetAccess | Capabilities.EnterpriseAuthentication | Capabilities.PrivateNetworkAccess;
         public override bool IsHidden => _isHidden;
         public override bool IsSelfContained => !_isBehaviorOnly;
-
-        public ServiceFunction(IService parentService, DPath theNamespace, string name, string localeSpecificName, string description,
-            DType returnType, BigInteger maskLambdas, int arityMin, int arityMax, bool isBehaviorOnly, bool isAutoRefreshable, bool isDynamic, bool isCacheEnabled, int cacheTimetoutMs, bool isHidden,
-            Dictionary<TypedName, List<string>> parameterOptions, ServiceFunctionParameterTemplate[] optionalParamInfo, ServiceFunctionParameterTemplate[] requiredParamInfo,
-            Dictionary<string, Tuple<string, DType>> parameterDefaultValues, string actionName = "", bool numberIsFloat = false, params DType[] paramTypes)
+        public bool IsPageable => !string.IsNullOrEmpty(_pageLink);
```

Comment:
>IsPageable [](http://example.com/codeflow?start=20&length=10)

Can you comment what the invariants are here?

For example, does Pageable mean that at the Fx level, it must return a TableValue?

Reply:
No. It means it implements x-ms-pageable and it must return a RecordValue